# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Coder-bot1/FlyRank-Internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Lane: Refresh / Content Opportunity Scoring**

I am choosing the Refresh / Content Opportunity Scoring lane because I want to investigate how data can help identify which content pages should be reviewed or refreshed first. The starter dataset contains observable signals such as impressions, sessions, clicks, CTR, average position, content age, freshness, and trend information. These signals can potentially help prioritize pages where a content review may be worthwhile.

I am choosing this lane provisionally because the internship allows the lane to be changed until the end of Week 4. For now, it looks useful because the problem connects directly to a practical decision: deciding which pages deserve limited content-review time first.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Research question:**

Can observable search and content signals help us rank which pages should be reviewed for a possible refresh first?

**Decision:**
Which pages should a content team prioritize for review?

**Action:**
A reviewer could inspect the highest-priority pages and decide whether they need actions such as updating outdated information, improving content depth, or investigating search-performance issues.

**Unit of analysis:**
One content page.

**Potential output:**
A ranked list of pages with an opportunity/risk score and reason codes explaining why each page was prioritized.

**Cost of a wrong recommendation:**
If a page is incorrectly prioritized, a reviewer may spend limited time improving a page that does not need attention. More importantly, if an important page is missed, the team may fail to investigate a genuine content or search-performance opportunity.

This is a decision-support problem rather than a claim that a model can guarantee that refreshing a page will improve its performance.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [13]:
!git clone https://github.com/flyrank-bih/flyrank-ml-internship-starter.git

fatal: destination path 'flyrank-ml-internship-starter' already exists and is not an empty directory.


In [14]:
import pandas as pd
import os

# Set the repository path
repo_path = "/content/flyrank-ml-internship-starter"

# Path to the starter dataset
data_path = os.path.join(
    repo_path,
    "data",
    "raw",
    "content_refresh_anonymized.csv"
)

# Check that the file exists
print("Dataset exists:", os.path.exists(data_path))
print("Dataset path:", data_path)

# Load the dataset
df = pd.read_csv(data_path)

# Display the first 5 rows
df.head()

Dataset exists: True
Dataset path: /content/flyrank-ml-internship-starter/data/raw/content_refresh_anonymized.csv


,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


In [15]:
print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))

print("\nColumns:")
print(df.columns.tolist())

Number of rows: 30000
Number of columns: 44

Columns:
['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


In [16]:
print("Number of pages:", len(df))

print("\nContent age:")
print("Minimum:", df["content_age_days"].min())
print("Median:", df["content_age_days"].median())
print("Maximum:", df["content_age_days"].max())

print("\n90-day impressions:")
print("Minimum:", df["impressions_90d"].min())
print("Median:", df["impressions_90d"].median())
print("Maximum:", df["impressions_90d"].max())

print("\n90-day sessions:")
print("Minimum:", df["sessions_90d"].min())
print("Median:", df["sessions_90d"].median())
print("Maximum:", df["sessions_90d"].max())

Number of pages: 30000

Content age:
Minimum: 90
Median: 236.0
Maximum: 564

90-day impressions:
Minimum: 1
Median: 731.0
Maximum: 517715

90-day sessions:
Minimum: 1
Median: 7.0
Maximum: 4345


In [17]:
print(df["trend_direction"].value_counts())

trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

My work will be able to provide observed and directional insights from the available page-level data. The dataset contains 30,000 pages, with content ages ranging from 90 to 564 days and a median age of 236 days. It also contains 90-day performance signals such as impressions and sessions, which can be used to identify patterns and help prioritize pages for human review.

The analysis may provide decision support by producing a ranked list of pages that appear more worthy of a content-refresh review based on observable signals. For example, the data contains 16,262 pages with a down trend direction, which provides a substantial group that could be investigated.

However, I cannot claim that a page will definitely improve after being refreshed. I cannot establish causality from this observational dataset, and I cannot claim that the analysis is "predicting Google" or predicting Google's ranking decisions. I also cannot claim that a model will generalize to all websites or future data without further validation.

The goal is therefore to investigate whether the available data can provide useful, reliable decision support for prioritizing content-review opportunities, rather than proving that a particular content change will cause better search performance.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.